### PR TITLE
Deleted tablets will not be displayed on Underreplicated Tablets page

### PR DESCRIPTION
**Pull request by omkaark**
This PR solves https://github.com/yugabyte/yugabyte-db/issues/15714

### CODE REVIEW

**Comment by omkaark** [comment]
@arybochkin can I please get a review!

File: src/yb/master/master-path-handlers.cc [hunk 1]
```diff
@@ -1612,9 +1612,10 @@ Result<std::vector<TabletInfoPtr>> MasterPathHandlers::GetUnderReplicatedTablets
 
   for (TabletInfoPtr t : nonsystem_tablets) {
     auto rm = t.get()->GetReplicaLocations();
+    bool is_deleted = t.get()->LockForRead()->is_deleted();
 
-    // Find out the tablets which have been replicated less than the replication factor
-    if (rm->size() < cluster_rf) {
+    // Find out the non-deleted tablets which have been replicated less than the replication factor
```
Comment (arybochkin) [inline src/yb/master/master-path-handlers.cc:1617]: nit: please put '.' at the end of the sentence.

**Review by arybochkin — APPROVED** [review]
LGTM too.

File: src/yb/master/master-path-handlers.cc (hunk 1 above)
Comment (omkaark) [inline src/yb/master/master-path-handlers.cc:1617]:
```suggestion
    // Find out the non-deleted tablets which have been replicated less than the replication factor.
```

**Comment by omkaark** [comment]
@hulien22 @lingamsandeep ping for review/merge!

**Review by hulien22 — APPROVED** [review]
LGTM @omkaark
Don't think you need all of our approvals btw - but I think @SrivastavaAnubhav is running your pr through our internal builds + tests so just need to wait for that

**Comment by omkaark** [comment]
Gotcha, thank you for letting me know!

**Pull request merged**